### DISCUSSION TITLE
Wrapping an image over an image convexly

### DISCUSSION THREAD

**Discussion by ankt-shrma** [ImageMagick/ImageMagick#2444] [Help] [answered]
**What I want**: When I apply an image over the pillow texture the artwork of the image should wrap convexly on the pillow.
Here is the link for the desired output that I need:
![desiredimage](https://user-images.githubusercontent.com/17960356/90733399-1f164800-e2ea-11ea-9824-c8e8b9e96439.png)

**Note**: In the above image, you can see the image is wrapped around the corners and convex from the center.

**Problem**: But when I apply image over the pillow texture image artwork is not curved convexly.
Here is the output that I currently have:
![currentimage](https://user-images.githubusercontent.com/17960356/90733449-2d646400-e2ea-11ea-8ec3-99c0a2cb9a45.png)

Also, for reference,
Here is the product image that I am using:
![Productimage](https://user-images.githubusercontent.com/17960356/90733559-584eb800-e2ea-11ea-8b53-49dbb568076d.png)

And here is the artwork I am overlaying it with:
![amit](https://user-images.githubusercontent.com/17960356/90733669-80d6b200-e2ea-11ea-858f-676748d116c7.jpg)

The desired solution would be that it should be working for any design I am overlaying it with.

Also, in the product image linked above, as you can see, there is some distortion in the top left corner, if I am overlaying it with any design, how can I achieve that distortion as well?

I am new to imagemagick, so are there any good resources to learn from and practice with?

Thank you.

**Comment by ankt-shrma** [comment]
It would be preferable to change only the shape of the design according to the shape of the product as mentioned above, as merging of the two images is being done using imagemagick's node.js library in my project.

**Comment by snibgo** [comment, marked as answer]
The example you show is a "pincushion", more or less, except for the rounded corners. See https://www.imagemagick.org/script/command-line-options.php#distort "-distort Barrel". You can tweak the ABCD parameters until you get the desired distortion.

My pages at http://im.snibgo.com have a number of solutions for distorting arbitrary shapes into other shapes.

**Comment by ankt-shrma** [comment]
@snibgo , thank you for the reply, that worked for me, also, I wanted to ask, sorry I am new to imagemagick, as I mentioned, if you see in the product image, provided above, in the "top left corner" there is some distortion(which is the case with any piece of clothing), if in the design that I am overlaying it with, I want to achieve that same distortion, how can I do that alongside "pincushioning"?

**Reply by DOEHOONLEE** [reply]
Hi! @ankt-shrma did you get it to work?
